### ISSUE TITLE
How can I define the file encoding?

### ISSUE THREAD

**Issue by lucasloami** [refile/refile#192]
Hi, is it possible to define an encoding to CSV files inside the `Refile.store.read` function? In this case, it would be similar to the command `CSV.read(file, {encoding: "bom|utf-8"})` in which you can specify whatever encoding you want.

**Comment by janko-m** [comment]
Yes, it would probably make sense to have this. If we would be doing this, then I guess we should add it to `#open` as well, for consistency (will it always be possible?).

**Comment by lucasloami** [comment]
I think so. One of the biggest issues I'm currently facing using Refile is that I'm not able to identify the file encoding by doing the command `Refile.store.read`, as it returns a file with encoding ASCII-8BIT, which is not actually an encoding, is the ruby way to say "this is a binary file". So I have to make several tests to figure out what the encoding is. 

My suggestion is: Refile should be smart/flexible enough to allow us define the encoding for text files (.txt, .csv, .tsv, etc.). If it's not possible to read the file in the selected encoding, the gem should raise an error, so that developers can catch it and handle this situation in their apps. (the ruby CSV lib works in a similar way)

**Comment by jnicklas** [comment]
But why not just encode in whatever encoding you want after the fact?

``` ruby
Refile.store.read(id).force_encoding("utf-8")
```

I don't see how that's at all worse than:

``` ruby
Refile.store.read(id, encoding: "utf-8")
```

**Comment by jnicklas** [comment]
I looked into how IO does this. IO makes it possible to specify both external and internal encoding when calling `open`, it can also accept `encoding: "foo:bar"` as a shortcut. `IO.read` only documents that it can accept `:encoding`, but it seems that it can accept the other options as well. On instances of IO, the `IO#read` method does not accept an encoding option. I think it might be challenging to expect backends to implement this stuff, and I'm not sure if we can somehow implement it in Refile itself so that we don't have to offload it onto the backends.

This is the kind of thing that can get complicated fast, so I'd accept a PR for this which provides a *simple* solution to this. If that's not possible, you can always cast the strings to the appropriate encoding.
